ReSharper.FSharp/src/FSharp.Psi/src/Util/FSharpExpressionUtil.cs

```diff
@@ -35,5 +36,54 @@ public static class FSharpExpressionUtil
 
     public static readonly Func<IFSharpExpression, bool> IsSimpleValueExpressionFunc = IsSimpleValueExpression;
     public static readonly Func<IFSharpExpression, bool> IsLiteralExpressionFunc = IsLiteralExpression;
+
+    // TODO: change name
+    public static IFSharpExpression GetOuterMostParentExpression(this IFSharpExpression expression)
+    {
+      var currentExpr = expression;
+      while (true)
+      {
+        if (MatchExprNavigator.GetByClauseExpression(currentExpr) is { } matchExpr)
+        {
+          currentExpr = matchExpr;
+          continue;
+        }
+        
+        if (SequentialExprNavigator.GetByExpression(currentExpr) is { } seqExpr &&
+            seqExpr.ExpressionsEnumerable.Last() == currentExpr)
+        {
+          currentExpr = seqExpr;
+          continue;
+        }
+
+        if (IfExprNavigator.GetByConditionExpr(currentExpr) is { } ifExpr &&
+            ifExpr.ElseExpr != null)
+        {
+          currentExpr = ifExpr;
+          continue;
+        }
+
+        if (BinaryAppExprNavigator.GetByArgument(currentExpr) is { } binaryAppExpr)
```

Comment:
We likely need to be more cautious here. Consider this:
```fsharp
let _: int = 1 + ""
```

I'd also check if the logic is the same for piping. For some reason I think there might be some quirks with binary operators and type conversions.

In general, it's usually better to not provide quick fix in cases were we aren't sure we work correctly than breaking the user code in an unexpected way. It's better to gradually enable cases later than to ship it as soon as possible and to deal with a situation where you can't easily fix all the corner cases and don't want to disable the feature too.